LLMonFHIR/FHIRSummary/FHIRResourceSummary.swift

```diff
@@ -29,13 +29,44 @@ final class FHIRResourceSummary: Sendable {
         
         
         init?(_ description: String) {
```

Comment:
I think there is substantial room for improvement here; I am not a huge fan of the parsing logic focusing on the number of lines and all the nested work that might not be necessary for a high level parsing improvement. We should generally assume that we get two lines and do a best effort that is minimal. Assuming all the lengths and others is too finicky.

1. Split the description in its lines by looking at `\n` characters (without all the need to remove whitespace, new lines, and more)
2. If we have 2 or more: 1. Line is the title, all other ones are the description (other lines are just concatenated again using `\n`).
3. If we have 1 line: That's the description and title
4. If we have 0 lines (empty string) we return nil and the outside function has to handle this (second comment).

Reply:
Agreed, keep it straightforward and simple, with a 1-2 retries